### DISCUSSION TITLE
How to use Fusion Cache to work with list queries?

### DISCUSSION THREAD

**Discussion by simonguertin** [ZiggyCreatures/FusionCache#246] [Q&A]
I am creating a LIST api on one of my resource (page index, pageSize, and query search parameters) Is it possible for FusionCache to work with that ? Cache results? and Invalidate the cache when some items are changed using different APIs ? 
For example:
/products?page=1&pageSize=5
/products/{id} details (GET/PUT)
Thanks!

**Comment by simonguertin** [comment]
I am using this, will fucsionCache invalidate redis if some of the items are added or removed somehow using other APIs ?: 
var contentList = _cache.GetOrSet<IEnumerable<Content>>(
            $"contentList:{contentType}_{page}_{pageSize}",
                _ => {
                    var query = db.Content
                    .OrderBy(b => b.Id)
//APPLY PAGESize/ page in Where
                    .ToList();
                return query;
            }

**Comment by jodydonetti** [comment]
Hi @simonguertin , and thanks for using FusionCache!

The cache key your are building for the cached page looks good to me. Having said that, no, there's currently no way to automatically evict a set of cache entries based on a change on one item which is part of another list stored as another cached entry.

What I would suggest is to cache the paged results for a lower amount of time.
